### PR TITLE
Bugs found

### PR DESCRIPTION
**Pull request by sannanansari**
Changes in  bugs finding

### CODE REVIEW

**Comment by sannanansari** [comment]
"All checks have failed" please you can explain about that. Please Review it.

**Comment by sannanansari** [comment]
As I want to save data. So I save whatever is print on screen.
result_data[channel] = {"node": node_id, "version": readable_version}
Channel = key
But actually you about remaining.
Basically I have save whatever is printed on screen

**Comment by Dexterp37** [comment]
Hi @sannanansari ! Just to get a better idea, as Georg mentioned, is this related to #13 ? If not, what is the problem you're trying to fix here? Thanks!

**Comment by sannanansari** [comment]
#13  @Dexterp37  you have comment to:-
# Change transform method so it can count number of probes per type, per channel and returns a structure with the data.
I was working in this way just to save data and return data.
So I just save whatever having being printing on Screen.

**Comment by Dexterp37** [comment]
@sannanansari I see! I can't find where you are counting in the PR you just sent. Was it left out? I'm dropping a few comments on it anyway!

File: probe_scraper/transform_probes.py [hunk 1]
```diff
@@ -156,10 +156,8 @@ def sorted_node_lists_by_channel(node_data):
                 'node_id': node_id,
                 'version': data['version'],
             })
-
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: Let's not remove blank lines :)

File: probe_scraper/transform_probes.py [hunk 2]
```diff
@@ -156,10 +156,8 @@ def sorted_node_lists_by_channel(node_data):
...
     for channel, data in channels.iteritems():
         channels[channel] = sorted(data, key=lambda n: int(n["version"]), reverse=True)
-
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: Let's not remove blank lines :)

File: probe_scraper/transform_probes.py [hunk 3]
```diff
@@ -172,19 +170,18 @@ def transform(probe_data, node_data, break_by_channel):
            release channel.
     """
     channels = sorted_node_lists_by_channel(node_data)
-
-    result_data = {}
+    result_data =defaultdict(dict)
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: Why did you change this part?

File: probe_scraper/transform_probes.py [hunk 4]
```diff
@@ -172,19 +170,18 @@ def transform(probe_data, node_data, break_by_channel):
...
     for channel, channel_data in channels.iteritems():
-        print "\n" + channel + " - transforming probe data:"
+        print "\n" + str(channel) + " - transforming probe data:"
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: Why did you change this part?

File: probe_scraper/transform_probes.py [hunk 5]
```diff
@@ -172,19 +170,18 @@ def transform(probe_data, node_data, break_by_channel):
...
         for entry in channel_data:
             node_id = entry['node_id']
             readable_version = entry["version"]
             print "  from: " + str({"node": node_id, "version": readable_version})
             for probe_type, probes in probe_data[channel][node_id].iteritems():
                 # Group the probes by the release channel, if requested
+                result_data[probe_type][channel] = {"node": node_id, "version": readable_version} 
                 extract_node_data(node_id, channel, probe_type, probes, result_data,
                                   readable_version, break_by_channel)
-
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: Let's not remove blank lines :)

File: probe_scraper/transform_probes.py [hunk 6]
```diff
@@ -172,19 +170,18 @@ def transform(probe_data, node_data, break_by_channel):
...
     for channel, channel_data in channels.iteritems():
-        print "\n" + channel + " - transforming probe data:"
+        print "\n" + str(channel) + " - transforming probe data:"
         for entry in channel_data:
             node_id = entry['node_id']
             readable_version = entry["version"]
             print "  from: " + str({"node": node_id, "version": readable_version})
             for probe_type, probes in probe_data[channel][node_id].iteritems():
                 # Group the probes by the release channel, if requested
+                result_data[probe_type][channel] = {"node": node_id, "version": readable_version}
```
Comment (Dexterp37) [inline probe_scraper/transform_probes.py]: You can create a new dictionary instead of reusing `result_data` and just use the new one for keeping track/counting the probes.

File: probe_scraper/transform_probes.py (hunk 3 above)
Comment (sannanansari) [reply probe_scraper/transform_probes.py]: I think that default dict is perfect because it won't give key error.

File: probe_scraper/transform_probes.py (hunk 4 above)
Comment (sannanansari) [reply probe_scraper/transform_probes.py]: I have just change it. Not any Reason behind it.

File: probe_scraper/transform_probes.py (hunk 6 above)
Comment (sannanansari) [reply probe_scraper/transform_probes.py]: OK. I will work on it.

Comment (sannanansari) [reply probe_scraper/transform_probes.py]: OK. Means I have Done Wrong this line.
**result_data[probe_type][channel] = {"node": node_id, "version": readable_version}**
You can Review it.

File: probe_scraper/transform_probes.py (hunk 4 above)
Comment (Dexterp37) [reply probe_scraper/transform_probes.py]: In general, if there isn't any specific reason to change a line, then it's better to leave it alone :) That makes it less likely to add new bugs to the code!

File: probe_scraper/transform_probes.py (hunk 6 above)
Comment (Dexterp37) [reply probe_scraper/transform_probes.py]: @sannanansari , I would expect this line to become something like:

```
probe_counts[channel][probe_type] += 1
```

**Comment by Dexterp37** [comment]
Let's close this, it appears you pushed another PR (#57 ). Let's take the conversation there.
